src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -128,7 +128,11 @@ void SoundRecorder::stop()
     if (m_isCapturing)
     {
         m_isCapturing = false;
-        m_thread.wait();
+
+        if (m_thread.has_value() && m_thread->joinable())
+        {
+            m_thread->join();
+        }
```

Comment:
> This is used twice, could probably be a separate function awaitCapturingThread().

Will do.

> Also, would it make sense to empty m_thread after join(), since it has now finished running?

There's no real reason to do so, `.emplace` will replace the current one anyway.

Reply:
> There's no real reason to do so, .emplace will replace the current one anyway.

We wouldn't even need this with the suggestion here: https://github.com/SFML/SFML/pull/1863#discussion_r762568259